src/lib/layout/Footer.svelte

```diff
@@ -0,0 +1,144 @@
+<script lang="ts">
+  import { onMount } from 'svelte';
+  import Icon from '$lib/component/Icon.svelte';
+  import Icons from '$lib/component/Icons';
+
+  function copyToClipboard(content: string) {
+    navigator.clipboard.writeText(content);
+  }
+
+  onMount(() => {
+    const iconsContainer: NodeListOf<Element> | null = document.querySelectorAll('.footer-icons');
+    if (iconsContainer !== null) {
+      iconsContainer.forEach((container) => {
+        const icons: HTMLCollection = container.children;
+
+        for (let i of icons) {
+          i.addEventListener('mouseover', (e) => {
+            const el = e.target as HTMLElement;
+            if (el !== null) {
+              el.style.opacity = '1.0';
+              for (let j of icons) {
+                if (j !== el) {
+                  (j as HTMLElement).style.opacity = '0.5';
+                }
+              }
+            }
+          });
+
+          i.addEventListener('mouseleave', (e) => {
+            const el = e.target as HTMLElement;
+            if (el !== null) {
+              el.style.opacity = '1.0';
+              for (let j of icons) {
+                if (j !== el) {
+                  (j as HTMLElement).style.opacity = '1.0';
+                }
+              }
+            }
+          });
+        }
+      });
+    }
+  });
```

Comment:
I think the readability here could be improved. Here are a few suggestions:
- Create a function above to change the opacity of the style, it could take the opacity as a param since it's the only difference in the event listeners.
- I'm not sure about this one, so could you try removing the `if (iconsContainer !== null)` and changing the for to `iconsContainer?.forEach`. I think it's the same thing and it reduces nesting

Reply:
This has been refactored by making the `addEventListener` a separate function and passing it all of the relevant variables. Please check it out as I believe that is not what you quite intended.

Reply:
I think it's clean but didn't the behaviour change too?

Reply:
I think it's clean. Just one thing, why do we need to the remaining icons' opacities in every event?

Reply:
makes sense now